### DISCUSSION TITLE
Compile error with V2.1.3 and PlatformIO - ESP32

### DISCUSSION THREAD

**Discussion by gerthalberg** [davetcc/tcMenu#164] [Q&A] [answered]
Hi, I get the following compile error when trying to compile a simple example. My thought is, that someone before me has experienced it.

> Executing task: platformio run <

Processing esp-wrover-kit (platform: espressif32; board: az-delivery-devkit-v4; framework: arduino)
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/az-delivery-devkit-v4.html
PLATFORM: Espressif 32 (3.3.2) > AZ-Delivery ESP-32 Dev Kit C V4
HARDWARE: ESP32 240MHz, 520KB RAM, 16MB Flash
DEBUG: Current (esp-prog) External (esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES: 
 - framework-arduinoespressif32 3.10006.210326 (1.0.6) 
 - tool-esptoolpy 1.30100.210531 (3.1.0) 
 - toolchain-xtensa32 2.50200.97 (5.2.0)
LDF: Library Dependency Finder -> http://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 32 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <TFT_eSPI> 2.3.70
|   |-- <SPIFFS> 1.0
|   |   |-- <FS> 1.0
|   |-- <FS> 1.0
|   |-- <SPI> 1.0
|-- <tcMenu> 2.1.3
|   |-- <IoAbstraction> 2.0.5
|   |   |-- <Wire> 1.0.1
|   |   |-- <TaskManagerIO> 1.2.3
|-- <TaskManagerIO> 1.2.3
|-- <IoAbstraction> 2.0.5
|   |-- <Wire> 1.0.1
|   |-- <TaskManagerIO> 1.2.3
|-- <SPI> 1.0
Building in release mode
Compiling .pio/build/esp-wrover-kit/src/TCMenu_eSPI_menu.cpp.o
Compiling .pio/build/esp-wrover-kit/src/tcMenuTfteSpi.cpp.o
Archiving .pio/build/esp-wrover-kit/libba0/libFS.a
Indexing .pio/build/esp-wrover-kit/libba0/libFS.a
Archiving .pio/build/esp-wrover-kit/lib52e/libSPIFFS.a
Indexing .pio/build/esp-wrover-kit/lib52e/libSPIFFS.a
Archiving .pio/build/esp-wrover-kit/lib261/libTFT_eSPI.a
Indexing .pio/build/esp-wrover-kit/lib261/libTFT_eSPI.a
Compiling .pio/build/esp-wrover-kit/lib117/Wire/Wire.cpp.o
src/tcMenuTfteSpi.cpp: In member function 'void TfteSpiDrawable::fontPtrToNum(const void*, int)':
src/tcMenuTfteSpi.cpp:102:44: error: ISO C++ forbids declaration of 'type name' with no type [-fpermissive]
         tft->setFreeFont(static_cast<const GFXfont *>(font));
                                            ^
src/tcMenuTfteSpi.cpp:102:44: error: expected '>' before 'GFXfont'
src/tcMenuTfteSpi.cpp:102:44: error: expected '(' before 'GFXfont'
src/tcMenuTfteSpi.cpp:102:44: error: 'GFXfont' was not declared in this scope
src/tcMenuTfteSpi.cpp:102:53: error: expected primary-expression before '>' token
         tft->setFreeFont(static_cast<const GFXfont *>(font));
                                                     ^
Compiling .pio/build/esp-wrover-kit/libe5d/TaskManagerIO/SimpleSpinLock.cpp.o
Compiling .pio/build/esp-wrover-kit/libe5d/TaskManagerIO/TaskManagerIO.cpp.o
*** [.pio/build/esp-wrover-kit/src/tcMenuTfteSpi.cpp.o] Error 1
============================================================================================= [FAILED] Took 2.88 seconds =============================================================================================
The terminal process "platformio 'run'" terminated with exit code: 1.

**Comment by davetcc** [comment]
Please provide the version of the designer, tcMenu library and plugins. Depending on the version of the designer UI, these can either be found from General Settings or About.

**Reply by gerthalberg** [reply]
Hi Dave,
The Menu designer is:
2.1.13
and the plug-ins 
<img width="601" alt="image" src="https://user-images.githubusercontent.com/15729261/133836139-8bfbe900-4bf3-4b34-956b-535406e1a18d.png">
and PlatformIO library is:
name=tcMenu
version=2.1.3
maintainer=www.thecoderscorner.com
author=davetcc
category=Other

**Comment by davetcc** [comment, marked as answer]
There is a bug in the 2.1.x plugins that included the TFT_eSPI.h header in the wrong case. On some platforms this caused problems. Open the file `tcMenuTfteSpi.h` and check that the include of the TFT_eSPI library looks exactly as follows:

        #include <TFT_eSPI.h>

If not change the case, originally the TFT part had a lower case 'f' instead of capital.

This is already fixed in the 2.2 BETA, and will be released soon.

EDIT- I've checked the Esp32Amplifier example on my hardware just now, all compiles and works as expected on 2.2.
